### DISCUSSION TITLE
[RR7] Possibility to name routes

### DISCUSSION THREAD

**Discussion by 4FunAndProfit** [remix-run/react-router#12244] [Proposals]
When you have something like this :  
```
export const routes: RouteConfig = [
  index("./home.tsx"),
  route("about", "./about.tsx"),

  layout("./auth/layout.tsx", [
    route("login", "./auth/login.tsx"),
    route("register", "./auth/register.tsx"),
  ]),

  ...prefix("concerts", [
    index("./concerts/home.tsx"),
    route(":city", "./concerts/city.tsx"),
    route("trending", "./concerts/trending.tsx"),
  ]),
];
``` 

It's common to have to use routes elsewhere in the app, but if they're heavily embedded in prefixes etc. it's not necessarily easy to use.

the ideal solution would be to name these routes so that you can easily access their properties by adding a third property
```
route(“trending”, “./concerts/trending.tsx”, “trendingConcerts”),
```
and perhaps retrieve the route's actual path by entering `routes[“trendingConcerts”]` => `/concerts/trending` 

What you think about it?

**Comment by pingusama1** [comment]
There are this discussion : https://github.com/remix-run/react-router/discussions/11951
